vitrina/datasets/views.py

```diff
@@ -182,12 +191,19 @@ def get(self, request, *args, **kwargs):
         return super(DatasetCreateView, self).get(request, *args, **kwargs)
 
     def form_valid(self, form):
-        object = form.save(commit=False)
-        object.slug = slugify(object.title)
-        return super().form_valid(form)
-
-
-class DatasetUpdateView(LoginRequiredMixin, PermissionRequiredMixin, UpdateView):
+        self.object = form.save(commit=False)
+        self.object.slug = slugify(self.object.title)
+        self.object.save()
+        set_comment(Dataset.CREATED)
+        return HttpResponseRedirect(self.get_success_url())
+
+
+class DatasetUpdateView(
+    RevisionMixin,
+    LoginRequiredMixin,
+    PermissionRequiredMixin,
```

Comment:
```suggestion
    LoginRequiredMixin,
    PermissionRequiredMixin,
    RevisionMixin
```

Permission checks should always go first, adjust the same on other views, where `RevisionMixin` is used.